test/shapiro_wilk.jl

```diff
@@ -0,0 +1,138 @@
+using HypothesisTests, LinearAlgebra, Test, Random
+using StableRNGs
+
+@testset "Shapiro-Wilk" begin
+    @testset "ShapiroWilkCoefs" begin
+        @test HypothesisTests.ShapiroWilkCoefs(3).A == [sqrt(2.0) / 2.0]
+        @test length(HypothesisTests.ShapiroWilkCoefs(3)) == 3
+
+        swc = HypothesisTests.ShapiroWilkCoefs(10)
+        @test length(swc) == 10
+        @test lastindex(swc) == 10
+        @test length(swc.A) == 5
+
+        @test swc.A[4] == swc[4] == -swc[7]
+        @test swc.A[2] == swc[2] == -swc[9]
+
+        swc = HypothesisTests.ShapiroWilkCoefs(11)
+        @test length(swc) == 11
+        @test lastindex(swc) == 11
+        @test length(swc.A) == 5
+
+        @test swc[6] == 0.0
+        @test swc.A[5] == swc[5] == -swc[7]
+        @test swc.A[3] == swc[3] == -swc[9]
+        @test swc.A[1] == swc[1] == -swc[11]
+
+        #anti-symmetry
+        swc = HypothesisTests.ShapiroWilkCoefs(20)
+        @test all([swc[i] == -swc[end - i + 1] for i in eachindex(swc)])
+
+        # Values obtained by calling `_swilkfort` fortran subroutine directly.
+
+        swc10 = HypothesisTests.ShapiroWilkCoefs(10)
+        res = swc10.A .- [0.573715, 0.32897, 0.214349, 0.122791, 0.0400887]
+        @test norm(res, 1) ≈ 0.0 atol = length(swc10) * eps(Float32)
+
+        swc20 = HypothesisTests.ShapiroWilkCoefs(20)
+        res = swc20.A .-
+              [0.473371, 0.32174, 0.255663, 0.208297, 0.16864, 0.133584, 0.101474,
+               0.0712893, 0.0423232, 0.0140351]
+        @test norm(res, 1) ≈ 0.0 atol = length(swc20) * eps(Float32)
+
+        rng = StableRNG(0x5bca7c69b794f8ce)
+        X = sort(rand(rng, Float64, 10))
+        # W, pval, _ = swilkfort(X)
+        W = 0.9088434774710951
+        # pval = 0.2731410626084226
+        @test HypothesisTests.unsafe_swstat(X, swc10) ≈ W atol = eps(Float32)
+    end
+
+    @testset "Shapiro-Wilk" begin
+        # syntactic tests
+
+        @test_throws ArgumentError ShapiroWilkTest([1, 2])
+        @test_throws ArgumentError("at least 3 samples are required, got 2") ShapiroWilkTest([1, 2],
+                                                   HypothesisTests.ShapiroWilkCoefs(3))
```

Comment:
Alignment is weird. Ideally we could put more on the second line, but given we can't maybe better keep everything on one line?
```suggestion
        @test_throws ArgumentError("at least 3 samples are required, got 2") ShapiroWilkTest([1, 2], HypothesisTests.ShapiroWilkCoefs(3))
```